pkg/systemd/services/service-details.jsx

```diff
@@ -319,6 +332,45 @@ export class ServiceDetails extends React.Component {
                 });
     }
 
+    deleteAction() {
+        // Stop timer so we don't get race conditions when the unit is gone.
+        if (this.interval)
+            clearInterval(this.interval);
```

Comment:
componentWillUnmount() calls clearInterval() as well. Either clearInterval() is idempotent, then the `if` condition here and in componentWillUnmount() are unnecessary. Or it's not, then both places should set `this.interval = null` afterwards.